rollup.config.js

```diff
@@ -1,62 +1,46 @@
-import commonjs from '@rollup/plugin-commonjs';
-import resolve from '@rollup/plugin-node-resolve';
 import babel from '@rollup/plugin-babel';
-import { sizeSnapshot } from 'rollup-plugin-size-snapshot';
+import typescript from '@rollup/plugin-typescript';
 import svgr from '@svgr/rollup';
-import path from 'path';
+import { defineConfig } from 'rollup'
 
-// eslint-disable-next-line import/extensions
 import pkg from './package.json';
 
 const extensions = ['.js', '.ts', '.tsx'];
 
-const defaultConfig = {
-  input: 'src/index.ts',
+const defaultOutputOptions = {
+  dir: 'dist',
+  exports: 'named',
+};
+
+export default defineConfig({
+  input: [
+    'src/index.ts',
+    'src/constants/index.ts',
+    'src/icons/index.ts',
+    'src/utils/index.ts',
+  ],
+  output: [
+    {
+      ...defaultOutputOptions,
+      format: 'cjs',
+      entryFileNames: '[name].cjs',
+    },
+    {
+      ...defaultOutputOptions,
+      format: 'esm',
+    },
+  ],
+  preserveModules: true,
   plugins: [
     svgr({
       expandProps: 'end',
     }),
-    resolve({
-      extensions,
-      moduleDirectories: [path.resolve(__dirname, '.'), 'node_modules'],
-    }),
-    commonjs({
-      include: 'node_modules/**',
-    }),
+    typescript(),
     babel({
-      babelHelpers: 'bundled', // TODO: Investigate 'runtime' option
+      babelHelpers: 'runtime',
       extensions,
       exclude: 'node_modules/**',
     }),
-    sizeSnapshot(),
   ],
-  /**
-   * @emotion/styled/base needs to be added explicitly because our
-   * @emotion/styled imports transpile specifically to @emotion/styled/base,
-   * which results in build errors if we do not include it.
-   */
-  external: [...Object.keys(pkg.dependencies), '@emotion/styled/base'],
-};
-
-export default [
-  {
-    ...defaultConfig,
-    output: [
-      {
-        file: 'dist/bundle.js',
-        format: 'cjs',
-        name: 'radianceUi',
-      },
-    ],
-  },
-  {
-    ...defaultConfig,
-    preserveModules: true,
-    output: [
-      {
-        dir: 'dist/bundle-es',
-        format: 'esm',
-      },
-    ],
-  },
-];
+  external: Object.keys(pkg.dependencies).map((name) => new RegExp(`^${name}`)),
```

Comment:
Is this to catch nested packages like `@emotion/styled/base`? I think a code comment would help clarify why we're setting it up this way.

Reply:
Yeah, exactly. The babel helpers are the same way. I will add a comment